di_test.go

```diff
@@ -77,9 +78,26 @@ func (suite *TestSuite) TestRegisterBeanAfterContainerInitialization() {
 	}
 	overwritten, err = RegisterBeanFactory("", Singleton, nil)
 	assert.False(suite.T(), overwritten)
-	if assert.Error(suite.T(), err) {
-		assert.Equal(suite.T(), expectedError, err)
-	}
+	assert.Error(suite.T(), err)
+	assert.Equal(suite.T(), expectedFactoryError, err)
```

Comment:
Can we actually move it to a separate test? For example, `TestRegisterBeanFactoryAfterContainerInitialization()`.